ios/BT/API/Model/DownloadedPackage.swift

```diff
@@ -2,35 +2,44 @@ import Foundation
 import ZIPFoundation
 import CryptoKit
 
-protocol DownloadedPackage {
-  func writeSignatureEntry(toDirectory directory: URL, filename: String) throws -> URL
-  func writeKeysEntry(toDirectory directory: URL, filename: String) throws -> URL
-}
-
-struct DownloadedPackageImpl: DownloadedPackage {
+protocol DownloadableFile {
 
-  init(keysBin: Data, signature: Data) {
-    bin = keysBin
-    self.signature = signature
-  }
+  static func create(from data: Data) -> Self?
+}
 
-  init?(compressedData: Data) {
-    guard let archive = Archive(data: compressedData, accessMode: .read) else {
+class DownloadedPackage: DownloadableFile {
```

Comment:
`final` class 🤓

Reply:
I had to make it a class to be able to inherit a mock class from it, thas why is not final